pkg/sources/github/feedlet.go

```diff
@@ -0,0 +1,438 @@
+	sc := t.servingclientset.ServingV1alpha1().Services(t.feedNamespace)
+
+	// First, check if deployment exists already.
+	if _, err := sc.Get(serviceName, metav1.GetOptions{}); err != nil {
+		if !apierrs.IsNotFound(err) {
+			glog.Infof("services.Get for %q failed: %s", serviceName, err)
+			return err
+		}
+		glog.Infof("service %q already deleted", serviceName)
+		return nil
+	}
+
+	return sc.Delete(serviceName, &metav1.DeleteOptions{})
+}
+
+func (t *GithubEventSource) deleteWebhook(trigger sources.EventTrigger, feedContext sources.FeedContext) error {
+	glog.Infof("Stopping github webhook feed with context %+v", feedContext)
+
+	components := strings.Split(trigger.Resource, "/")
+	owner := components[0]
+	repo := components[1]
+
+	webhookID, err := stringFrom(feedContext.Context, webhookIDKey)
+	if err != nil {
+		return fmt.Errorf("Failed to get webhook ID from context: %v", err)
+	}
+
+	accessToken, err := stringFrom(trigger.Parameters, accessTokenKey)
+	if err != nil {
+		return fmt.Errorf("Failed to get access token from trigger parameters: %v", err)
+	}
+
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: accessToken},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+	client := ghclient.NewClient(tc)
+
+	id, err := strconv.ParseInt(webhookID, 10, 64)
+	if err != nil {
+		glog.Warningf("Failed to convert webhook %q to int64 : %s", webhookID, err)
+		return err
+	}
+	r, err := client.Repositories.DeleteHook(ctx, owner, repo, id)
+	if err != nil {
+		if errResp, ok := err.(*ghclient.ErrorResponse); ok {
+			// If the webhook doesn't exist, nothing to do
+			if errResp.Message == "Not Found" {
+				glog.Info("Webhook doesn't exist, nothing to delete.")
+				return nil
+			}
+		}
+		glog.Warningf("Failed to delete the webhook: %s", err)
+		glog.Warningf("Response:\n%+v", r)
+		return err
+	}
+	glog.Infof("Deleted webhook %q successfully", webhookID)
+	return nil
+}
+
+func (t *GithubEventSource) createWebhook(trigger sources.EventTrigger, domain string) (*sources.FeedContext, error) {
+
+	glog.Infof("CREATING GITHUB WEBHOOK")
+
+	accessToken, err := stringFrom(trigger.Parameters, accessTokenKey)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get access token from trigger parameters: %v", err)
+	}
+
+	secretToken, err := stringFrom(trigger.Parameters, secretTokenKey)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get secret token from trigger parameters: %v", err)
+	}
+
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: accessToken},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+
+	client := ghclient.NewClient(tc)
+	active := true
+	name := "web"
+	config := make(map[string]interface{})
+	config["url"] = fmt.Sprintf("http://%s", domain)
+	config["content_type"] = "json"
+	config["secret"] = secretToken
+	// TODO(n3wscott): GitHub has many types of events, we will have to make this dynamic when we support
```

Comment:
This seems like a fundamental feature to me. Right now, it doesn't matter what my event type is, I always get `pull_request` events. At the very least this limitation should be loudly documented.

Is there a plan for making this dynamic? How would it work?

Reply:
Yes! totally agree. I think this is a huge issue that will take some re-engineering and might be too large for this PR.